src/js/lib/content/src/templates.ts

```diff
@@ -0,0 +1,245 @@
+/// This module defines helper methods for templates view.
+
+import { ProjectManager } from './project_manager'
+
+const PM = ProjectManager.default()
+
+const PROJECTS_LIST = 'projects-list'
+const PROJECTS_LIST_NEW_PROJECT = 'projects-list-new-project'
+
+const CARD_SPREADSHEETS = 'card-spreadsheets'
+const CARD_GEO = 'card-geo'
+const CARD_VISUALIZE = 'card-visualize'
+const CARD_BMW_DRIVERS = 'card-bmw-drivers'
+
+const ALL_CARDS = [
+    CARD_SPREADSHEETS,
+    CARD_GEO,
+    CARD_VISUALIZE,
+    CARD_BMW_DRIVERS
+]
+
+/**
+ * The sore for hidden elements.
+ *
+ * When the templates view is loaded, it hides some top-level elements, as
+ * their style is messing up scrolling. Hidden elements will be restored before
+ * loading the IDE.
+ */
+let hiddenElements: HTMLDivElement[] = []
+
+/** Status box div element for displaying errors. */
+let statusBox: HTMLElement = undefined
+
+/**
+ * Display the templates view.
+ *
+ * Main entry point. Loads the templates HTML markup, loads the projects list
+ * and sets callbacks on the template cards.
+ *
+ * @param openProject the callback that opens IDE with the provided project.
+ */
+async function loadTemplatesView(openProject: (project: string) => void): Promise<void> {
+    const templatesView = require('./templates-view.html')
+    hideRootHtml()
+    document.body.innerHTML += templatesView
+    statusBox = document.getElementById('templates-status-box')
+
+    try {
+        await loadProjectsList(openProject)
+    } catch (error) {
+        displayStatusBox("Failed to load projects.")
+    }
+
+    setTemplateCardHandlers(openProject)
+}
+
+/**
+ * Remove the top-level div elements from the scene.
+ */
```

Comment:
what is a "top-level div"?  These docs do not describe what this function is about.